Add basic SPM support

KaiOelfke opened the pull request:
I'm working on improving the SPM support for [Ensembles 2](https://www.ensembles.io/) and RNCryptor-objc is a dependency. This PR adds a Package.swift, a header file and some header symlinks so RNCryptor-objc can be used as a Swift package dependency in other Swift packages.

By default SPM uses the `include` directory for public headers. But it can be overwritten via `publicHeaderPath`, if preferred. According to my current understanding all public headers should be inside this directory plus the umbrella header. Internal headers in other places can be specified via `.cSettings` and `.headerSearchPath`.

Rspoon3 commented:
Any idea when this can be merged? I would also like SPM support.

raf-wtag commented:
I want to replace this library import to SPM from cocoapods in my project. Found that the Package.swift is missing. But this PR has been opened since 2 years!! Any idea when will be this changes merge?

rnapier commented:
I've tested this a bit, and it does work as expected. Generally I expect folks using SPM to pull it from the Swift version of RNCryptor rather than this one, but it doesn't seem to cause any problems in my tests.

The following configuration seems fine:

```
    dependencies: [
        .package(url: "https://github.com/RNCryptor/RNCryptor-objc", branch: "master")
    ],
```

And then in `targets`:

            dependencies: [
                .product(name: "RNCryptor", package: "RNCryptor-objc"),
            ]),

KaiOelfke commented:
FYI, I added this as Ensembles is implemented in Objective-C and was therefore using this. When I added SPM support for Ensembles I tried to use the same dependencies.

KaiOelfke commented:
@rnapier You could also create a new release git tag, if you have the time. Then people could configure their dependency with a proper semantic version instead of pinning the master branch.
